File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
// 1달 = 28일
// 1년 = 28 * 12 일

function solution(today, terms, privacies) {
    var answer = [];
    const types = new Map();
    
    const todayDays = convertToDays(today)
    
    // 1) terms에 있는거 map에 하나씩 넣기.
    for(const term of terms){
        const [type, month] = term.split(' ');
        types.set(type, month);
    }
    // 2) privacy 하나씩 검사.
    for(let i=0; i < privacies.length; i++){
        const [date, type] = privacies[i].split(' ');
        const termMonth = types.get(type);
        
        const privacyDays = convertToDays(date)
        
        console.log('@',termMonth*28)
        console.log(todayDays-privacyDays)
        
        if(termMonth*28 <= todayDays-privacyDays){
            answer.push(i+1);
        }
    }
    
    return answer;
}

function convertToDays(date){
    const [year, month, day] = date.split('.').map(Number);
    return year * 28 * 12 + month * 28 + day;
}

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    types = {}
    
    for term in terms:
        type, month = term.split(' ')
        types[type] = int(month) * 28
    
    for i, privacy in enumerate(privacies):
        date, type = privacy.split(' ')
        days = left_days(today, date)
        
        if days >= types[type]:
            answer.append(i+1)
        
    return answer

def left_days(a, b):
    year_a, month_a, day_a = list(map(int, a.split('.')))
    year_b, month_b, day_b = list(map(int, b.split('.')))
    
    return (year_a - year_b) * 12 * 28 + (month_a - month_b) * 28 + (day_a - day_b)